apps/src/maze/mazeController.js

```diff
@@ -0,0 +1,363 @@
+/**
+ * Blockly Apps: Maze
+ *
+ * Copyright 2012 Google Inc.
+ * http://blockly.googlecode.com/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview JavaScript for Blockly's Maze application.
+ * @author fraser@google.com (Neil Fraser)
+ */
+
+const timeoutList = require('../lib/util/timeoutList');
+
+const AnimationsController = require('./animationsController');
+const MazeMap = require('./mazeMap');
+const drawMap = require('./drawMap');
+const getSubtypeForSkin = require('./mazeUtils').getSubtypeForSkin;
+const tiles = require('./tiles');
+
+module.exports = class MazeController {
+  constructor(level, skin, config) {
+    const Type = getSubtypeForSkin(config.skinId);
+    const subtype = new Type(this, config);
+
+    this.stepSpeed = 100;
+
+    this.level = level;
+    this.skin = skin;
+    this.startDirection;
+
+    this.map;
+    this.subtype = subtype;
+    this.animationsController;
+    this.executionInfo;
+    this.store;
+
+    this.pegmanD;
+    this.pegmanX;
+    this.pegmanY;
```

Comment:
Does this appropriately add these properties to the object shape?  My understanding is that without an assignment this is a no-op.

Reply:
You're correct; I put these here for readability, not functionality. Do you have a preference for a more useful way to call out which properties are declared on a given instance?

Reply:
I'm not sure... can we set them to `null`/`undefined` for clarity?